ols/src/cache/redis_cache.py

```
@@ -70,14 +81,33 @@ def insert_or_append(self, user_id: str, conversation_id: str, value: str) -> No
             value: The value to set.
 
         Raises:
-            OutOfMemoryError: If item is evicted when Redis allocated
-                memory is higher than maxmemory.
+            RedisConnectionError: If item is unable to update after REDIS_MAX_RETRY.
         """
         key = super().construct_key(user_id, conversation_id)
-
         old_value = self.get(user_id, conversation_id)
-        with self._lock:
-            if old_value:
-                self.redis_client.set(key, old_value + "\n" + value)
-            else:
-                self.redis_client.set(key, value)
+        retry_count = 0
+        while retry_count < constants.REDIS_MAX_RETRY:
+            logger.debug("Updating redis cache ")
+            with self._lock:
+                if old_value:
+                    old_value.append(value)
+                    try:
+                        self.redis_client.set(key, old_value)
+                        break
+                    except Exception as redis_err:
+                        logger.error(
+                            f"Exception during updating the cache: {redis_err}"
+                        )
+                        retry_count += 1
+                else:
+                    values = [value]
+                    try:
+                        self.redis_client.set(key, values)
+                        break
+                    except Exception as redis_err:
+                        logger.error(
+                            f"Exception during creating the cache: {redis_err}"
+                        )
+                        retry_count += 1
+
+        raise RedisMaxRetryError("Updating cache failed after max retries")
```

this should also report the last exception/error we got